Gems/Atom/RHI/Code/Include/Atom/RHI/MultiDeviceDrawItem.h

```diff
@@ -0,0 +1,298 @@
+                {
+                    deviceShaderResourceGroup[i] = shaderResourceGroups[i]->GetDeviceShaderResourceGroup(deviceIndex).get();
+                }
+
+                drawItem.m_shaderResourceGroups = deviceShaderResourceGroup.data();
+            }
+        }
+
+        //! Unique SRG, not shared within the draw packet. This is usually a per-draw SRG, populated with the shader variant fallback
+        //! key
+        void SetUniqueShaderResourceGroup(const MultiDeviceShaderResourceGroup* uniqueShaderResourceGroup)
+        {
+            for (auto& [deviceIndex, drawItem] : m_deviceDrawItems)
+            {
+                drawItem.m_uniqueShaderResourceGroup = uniqueShaderResourceGroup->GetDeviceShaderResourceGroup(deviceIndex).get();
+            }
+        }
+
+        //! Array of root constants to bind (count must match m_rootConstantSize).
+        void SetRootConstants(const uint8_t* rootConstants, uint8_t rootConstantSize)
+        {
+            for (auto& [deviceIndex, drawItem] : m_deviceDrawItems)
+            {
+                drawItem.m_rootConstantSize = rootConstantSize;
+                drawItem.m_rootConstants = rootConstants;
+            }
+        }
+
+        //! List of scissors to be applied to this draw item only. Scissor will be restore to the previous state
+        //! after the MultiDeviceDrawItem has been processed.
+        void SetScissors(const Scissor* scissors, uint8_t scissorsCount)
+        {
+            for (auto& [deviceIndex, drawItem] : m_deviceDrawItems)
+            {
+                drawItem.m_scissorsCount = scissorsCount;
+                drawItem.m_scissors = scissors;
+            }
+        }
+
+        //! List of viewports to be applied to this draw item only. Viewports will be restore to the previous state
+        //! after the MultiDeviceDrawItem has been processed.
+        void SetViewports(const Viewport* viewports, uint8_t viewportCount)
+        {
+            for (auto& [deviceIndex, drawItem] : m_deviceDrawItems)
+            {
+                drawItem.m_viewportsCount = viewportCount;
+                drawItem.m_viewports = viewports;
+            }
+        }
+
+    private:
+        MultiDevice::DeviceMask m_deviceMask{ MultiDevice::DefaultDevice };
+        //! A map of all device-specific DrawItems, indexed by the device index
+        AZStd::unordered_map<int, DrawItem> m_deviceDrawItems;
+        //! A map of all device-specific IndexBufferViews, indexed by the device index
+        AZStd::unordered_map<int, IndexBufferView> m_deviceIndexBufferView;
+        //! A map of all device-specific StreamBufferViews, indexed by the device index
+        AZStd::unordered_map<int, AZStd::vector<StreamBufferView>> m_deviceStreamBufferViews;
+        //! A map of all device-specific ShaderResourceGroups, indexed by the device index
+        AZStd::unordered_map<int, AZStd::vector<ShaderResourceGroup*>> m_deviceShaderResourceGroups;
```

Comment:
Why do we need to cache these here? Maybe add a comment about that.

Reply:
Since the `DrawItem` only holds raw pointers to these three members, we once again need to cache the device-specific variants. 
I added comments for that.